### PR TITLE
Once more, with feeling

### PR DESCRIPTION
**Pull request by pdehaan**
Something still doesn't feel right. I don't think we're either writing the config file, or it's difficult to inspect w/ `docker run -it` possibly not running ./run.sh. Will need more debuggery

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,8 +9,8 @@ ARG TEST_CONFIG="{'error':'missing_config'}"
 ARG TEST_ENV="{'error': 'missing_test_env'}"
 
 RUN apt-get -y update && \
-    apt-get install -y -qq software-properties-common && \
-    apt-get install -y -qq wget unzip \ 
+    apt-get install -qq -y software-properties-common && \
+    apt-get install -qq -y wget unzip xvfb \
```
Comment (pdehaan) [inline Dockerfile]: Order of `-qq` seemingly doesn't matter. `apt-get install` is still noisy as hell.
But we're now installing xvfb, fwiw.

File: Dockerfile [hunk 2]
```diff
@@ -34,7 +34,7 @@ RUN unzip master.zip 'gecko-dev-master/testing/modules/*' -d /tmp && \
 RUN unzip master.zip 'gecko-dev-master/testing/tps/*' -d /tmp && \
     mv /tmp/gecko-dev-master/testing/tps /mnt/extra/mozilla-central/testing 
 
-RUN wget -O firefox-nightly.tar.bz2 'https://download.mozilla.org/?product=firefox-nightly-latest-ssl&os=linux64&lang=en-US' && \
+RUN wget -q -O firefox-nightly.tar.bz2 'https://download.mozilla.org/?product=firefox-nightly-latest-ssl&os=linux64&lang=en-US' && \
```
Comment (pdehaan) [inline Dockerfile]: This silences the download progress of Firefox Nightly, which is a welcome change from the default verbosity.

File: run.sh [hunk 3]
```diff
@@ -5,10 +5,9 @@ TEST_CONFIG=$1
 BINARY=/tests/firefox-nightly/firefox
 CONFIG_FILE=/tests/config.json
 
-echo ${TEST_CONFIG} > $CONFIG_FILE
+echo $TEST_CONFIG > $CONFIG_FILE
```
Comment (pdehaan) [inline run.sh]: I've been fiddling with this, and I'm not sure if I'm just not getting docker, or if I'm calling something wrong. Calling `docker run -e "TEST_ENV=prod" -e "TEST_CONFIG='{abd}'" -it 344b /bin/bash` doesn't seem to call the run.sh file, and our config.json file never gets created. There is one in venv, but it feels oddly default and not related to our work. But the file is created if I interactively /bin/bash into the folder and call run.sh. It's weird. Maybe we can look Mon/Tues of next week.

Comment (pdehaan) [reply run.sh]: To clarify slightly... if I rename my config file to /tests/confrigg.json, and run `docker run -e "TEST_CONFIG='{boo:yah}'" -it {sha} /bin/bash`, it seems like the ./run.sh file isn't executed and my confriggg.json isn't created, but there is still a /tests/venv/config.json file present with some default looking values (possibly from create_venv, i should look there next). Running ./run.sh from within the container creates the expected /tests/confriggg.json file. 🤷‍♀️

File: run.sh [hunk 4]
```diff
@@ -3,12 +3,11 @@ set -e
 
 TEST_CONFIG=$1
 BINARY=/tests/firefox-nightly/firefox
-CONFIG_FILE=/tests/config.json
+CONFIG_FILE=/tests/venv/config.json
```
Comment (rpappalax) [inline run.sh]: ugly hack

Comment (pdehaan) [reply run.sh]: This is super confusing...
It seems like create_venv.py is creating the dummy config.json file: https://github.com/mozilla/gecko-dev/blob/master/testing/tps/create_venv.py#L173

Not sure the easiest way to run our Docker container and confirm that our config is overriding that create_venv's auto-generated version.

Comment (rpappalax) [reply run.sh]: whoa, good find.  i've opted to go ahead and assume the config will live in the venv.  without changing gecko-dev, we can then overwrite this default file with our own and call it a day.

**Pull request merged**